lib/Widgets/Placeholder.vala

```diff
@@ -0,0 +1,167 @@
+            use_markup = true,
+            xalign = 0
+        };
+        description_label.add_css_class (Granite.STYLE_CLASS_H2_LABEL);
+
+        buttonbox = new Gtk.Box (Gtk.Orientation.VERTICAL, 0) {
+            visible = false
+        };
+
+        var grid = new Gtk.Grid ();
+        grid.attach (image, 0, 0, 1, 2);
+        grid.attach (title_label, 1, 0);
+        grid.attach (description_label, 1, 1);
+        grid.attach (buttonbox, 1, 2);
+        grid.set_parent (this);
+
+        halign = Gtk.Align.CENTER;
+        valign = Gtk.Align.CENTER;
+        hexpand = true;
+
+        bind_property ("title", title_label, "label");
+        bind_property ("description", description_label, "label");
+        bind_property (
+            "description", description_label, "visible", BindingFlags.SYNC_CREATE | BindingFlags.DEFAULT,
+            (binding, srcval, ref targetval) => {
+                var label = (string) srcval;
+                if (label != null) {
+                    targetval.set_boolean (true);
+                } else {
+                    targetval.set_boolean (false);
+                }
+
+                return true;
+            },
+            null
+        );
+
+        bind_property ("icon", image, "gicon");
+        bind_property (
+            "icon", image, "visible", BindingFlags.SYNC_CREATE | BindingFlags.DEFAULT,
+            (binding, srcval, ref targetval) => {
+                var gicon = (Icon) srcval;
+                if (gicon != null) {
+                    targetval.set_boolean (true);
+                } else {
+                    targetval.set_boolean (false);
+                }
+
+                return true;
+            },
+            null
+        );
+    }
+
+    ~Placeholder () {
+        get_first_child ().unparent ();
+    }
+
+    /**
+     * Appends new {@link Gtk.Button} to the placeholder's action area
+     *
+     * @param icon the {@link GLib.Icon} that describes this action
+     * @param label a {@link string} to use as the title for this action. It should use Title Case.
+     * @param description a {@link string} to use as a description for this action. It should use sentence case.
+     * @return a {@link Gtk.Button} representing this action
+     */
+    public Gtk.Button append_button (Icon icon, string label, string description) {
+        var image = new Gtk.Image.from_gicon (icon) {
+            icon_size = Gtk.IconSize.LARGE
+        };
+
+        var label_widget = new Gtk.Label (label) {
+            wrap = true,
+            xalign = 0
+        };
+        label_widget.add_css_class (Granite.STYLE_CLASS_H3_LABEL);
+
+        var description_widget = new Gtk.Label (description) {
+            wrap = true,
+            xalign = 0
+        };
+
+        var grid = new Gtk.Grid ();
+        grid.attach (image, 0, 0, 1, 2);
+        grid.attach (label_widget, 1, 0);
+        grid.attach (description_widget, 1, 1);
+
+        var button = new Gtk.Button () {
```

Comment:
Intentional tjhat there's no `.flat`?

Reply:
Yeah in SCSS I'm extending `.flat` here. Could just set it in the widget itself though! I don't feel super strongly about it

Reply:
That works too, though after libadwaita header bars I'm super conscious about not extending things if possible - the amount of generated css snowballs very very quickly with this.

Reply:
Good point. Switched to adding flat class here